lib/lte_link_control/lte_lc.c

```diff
@@ -1461,9 +1464,12 @@ int lte_lc_neighbor_cell_measurement(struct lte_lc_ncellmeas_params *params)
 			LTE_LC_NEIGHBOR_SEARCH_TYPE_GCI_EXTENDED_COMPLETE),
 		 "Invalid argument, API does not accept enum values directly anymore");
 
+	k_mutex_lock(&ncellmeas_ongoing_mutex, K_SECONDS(10));
```

Comment:
Should we print a warning trace in case of timeout?

Reply:
There is a warning now with semaphore implementation.